openbas-api/src/main/java/io/openbas/rest/team/TeamApi.java

```diff
@@ -236,4 +232,26 @@ public Team updateTeamUsers(@PathVariable String teamId, @Valid @RequestBody Upd
         team.setUsers(fromIterable(teamUsers));
         return teamRepository.save(team);
     }
+
+    // -- PRIVATE --
+
+    private void isTeamAlreadyExists(@NotNull final TeamCreateInput input) {
+        List<Team> teams = this.teamRepository.findAllByName(input.getName());
+        // 1. input is not contextual and a not contextual team already exist
+        if (FALSE.equals(input.getContextual()) && teams.stream().anyMatch(t -> FALSE.equals(t.getContextual()))) {
+            throw new UnsupportedOperationException("Global teams (non contextual) cannot have the same name (already exists)");
+        }
+        // 2. input is contextual and a contextual team on the same exercise or scenario already exist
+        if (TRUE.equals(input.getContextual())) {
+            String exerciseId = input.getExerciseIds().stream().findFirst().orElse(null);
+            if (hasText(exerciseId) && teams.stream().anyMatch(t -> TRUE.equals(t.getContextual()) && t.getExercises().stream().anyMatch((e) -> exerciseId.equals(e.getId())))) {
+                throw new UnsupportedOperationException("A contextual team with the same name already exists on this simulation");
+            }
+            String scenarioId = input.getScenarioIds().stream().findFirst().orElse(null);
+            if (hasText(scenarioId) && teams.stream().anyMatch(t -> TRUE.equals(t.getContextual()) && t.getScenarios().stream().anyMatch((e) -> scenarioId.equals(e.getId())))) {
+                throw new UnsupportedOperationException("A contextual team with the same name already exists on this simulation");
+            }
```

Comment:
"on this scenario"

Reply:
Maybe manage better this unsopportedOperationException (500 ->400, 409?). 
On the frontend, we have: 
![image](https://github.com/user-attachments/assets/a82998cd-020c-464a-a0bf-c01fee199d3f)